lib/multi.c

```diff
@@ -1208,10 +1303,49 @@ static CURLMcode Curl_multi_wait(struct Curl_multi *multi,
           mask |= CURL_WAIT_POLLOUT;
         if(r & POLLPRI)
           mask |= CURL_WAIT_POLLPRI;
+#endif
 
         extra_fds[i].revents = mask;
       }
 
+#ifdef USE_WINSOCK
+      /* Count up all our own sockets that had activity,
+         and remove them from the event. */
+      if(curlfds) {
+        data = multi->easyp;
+        while(data) {
+          bitmap = multi_getsock(data, sockbunch);
+
+          for(i = 0; i < MAX_SOCKSPEREASYHANDLE; i++) {
+            if(bitmap & (GETSOCK_READSOCK(i) | GETSOCK_WRITESOCK(i))) {
+              WSANETWORKEVENTS events = {0};
+              if(WSAEnumNetworkEvents(sockbunch[i], multi->wsa_event,
+                                      &events) == 0) {
+                if(events.lNetworkEvents != 0)
+                  retcode++;
+              }
+              if(ret && !events.lNetworkEvents && GETSOCK_WRITESOCK(i)) {
```

Comment:
This causes:
```
In file included from multi.c:35:
multi.c: In function 'Curl_multi_wait':
multiif.h:52:33: error: '<<' in boolean context, did you mean '<'? [-Werror=int-in-bool-context]
   52 | #define GETSOCK_WRITESOCK(x) (1 << (GETSOCK_WRITEBITSTART + (x)))
      |                              ~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
multi.c:1327:51: note: in expansion of macro 'GETSOCK_WRITESOCK'
 1327 |               if(ret && !events.lNetworkEvents && GETSOCK_WRITESOCK(i)) {
      |         
```

Reply:
…bit again by another bitwise thing! Fixed.